core/page.ml

```diff
@@ -160,7 +161,7 @@ module Make_RealPage (C : JS_PAGE_COMPILER) (G : JS_CODEGEN) = struct
       ~cgi_env
       ~body:printed_code
       ~html:(Value.string_of_xml ~close_tags:true bs)
-      ~head:(script_tag welcome_msg ^ "\n" ^ script_tag (C.primitive_bindings) ^ "\n" ^ script_tag("  var _jsonState = " ^ state_string ^ "\n" ^ init_vars)
+      ~head:(script_tag welcome_msg ^ "\n" ^ script_tag (C.primitive_bindings) ^ "\n" ^ script_tag("  var _jsonState = " ^ escaped_state_string ^ "\n" ^ init_vars)
```

Comment:
I think we should make `_jsonState` a function, because global variables never get garbage collected. I suspect the state can grow quite large.

Reply:
On second thought. I may not matter in this case, as it we are only going to have one global variable. The big string would need to be allocated one way or another anyways.